tests/spark_autologging/test_pyspark_ml_autologging.py

```diff
@@ -0,0 +1,222 @@
+import importlib
+import inspect
+import pytest
+from collections import namedtuple
+from unittest import mock
+
+import mlflow
+from mlflow.utils import (
+    truncate_dict,
+)
+from mlflow.utils.validation import (
+    MAX_PARAM_VAL_LENGTH,
+    MAX_ENTITY_KEY_LENGTH,
+)
+
+from tests.autologging.fixtures import test_mode_off
+from tests.spark_autologging.utils import spark_session  # pylint: disable=unused-import
+from pyspark.ml.linalg import Vectors
+from pyspark.ml.regression import LinearRegression
+from mlflow.pyspark.ml import _should_log_model, _get_instance_param_map, _log_model_allowlist
+
+MODEL_DIR = "model"
+MLFLOW_PARENT_RUN_ID = "mlflow.parentRunId"
+
+
+@pytest.fixture(scope="module")
+def dataset_binary(spark_session):
+    yield spark_session.createDataFrame(
+        [(1.0, Vectors.dense(1.0)),
+         (0.0, Vectors.sparse(1, [], []))] * 100,
+        ["label", "features"])
+
+
+@pytest.fixture(scope="module")
+def dataset_20_classes(spark_session):
+    label_list = list(range(20)) * 10
+    features_list = [Vectors.dense(1.0),
+                     Vectors.sparse(1, [], [])] * 100
+    yield spark_session.createDataFrame(
+        zip(label_list, features_list),
+        ["label", "features"])
+
+
+def truncate_param_dict(d):
+    return truncate_dict(d, MAX_ENTITY_KEY_LENGTH, MAX_PARAM_VAL_LENGTH)
+
+
+def stringify_dict_values(d):
+    return {k: str(v) for k, v in d.items()}
+
+
+def get_expected_class_tags(estimator):
+    return {
+        'estimator_name': estimator.__class__.__name__,
+        'estimator_class': estimator.__class__.__module__ + "." + estimator.__class__.__name__,
+    }
+
+
+def get_run_data(run_id):
+    client = mlflow.tracking.MlflowClient()
+    data = client.get_run(run_id).data
+    # Ignore tags mlflow logs by default (e.g. "mlflow.user")
+    tags = {k: v for k, v in data.tags.items() if not k.startswith("mlflow.")}
+    artifacts = [f.path for f in client.list_artifacts(run_id)]
+
+    RunData = namedtuple('RunData', ['params', 'metrics', 'tags', 'artifacts'])
+    return RunData(data.params, data.metrics, tags, artifacts)
+
+
+def load_model_by_run_id(run_id):
+    return mlflow.spark.load_model("runs:/{}/{}".format(run_id, MODEL_DIR))
+
+
+def get_dataset_binary(spark_session):
```

Comment:
```suggestion
def get_binary_classification_dataset(spark_session):
```

Can we renmae this function to?